### ISSUE TITLE
App crashed while decoding barcode

### ISSUE THREAD

**Issue by burakcolakappcent** [Dynamsoft/barcode-reader-mobile-samples#8]
Hi, we are using your SDK about 1 years.
But one is not seeming resolved since then.

Heap corruption detected, free list is damaged at ### *** Incorrect guard value: ###

We have this error for some device from day 1.
Could you please look at this and give us some idea that how can we solve ?

DynamsoftBarcodeReader
(Missing UUID

UUID's are not included sometimes also.

Regards.

**Comment by Dynamsoft-Jerson** [comment]
Hi, thanks for your feedback. Can you share more details?
1)iOS version
2)xcode version
3)device type
4)DynamsoftBarcodeReader SDK version
5)DynamsoftCameraEnhancer SDK version (If you used)
6)crash log

**Comment by burakcolakappcent** [comment]
1) 15.3.1, 15.4.0, 15.4.1, 14.6.0, 12.5.5, 14.8.1, 15.0.2, 13.6.1, 14.1.0, 15.2.1
2) Archived on Xcode version 13.3.1, these logs are from production firebase
3) iPhone 8 Plus, iPhone 12, iPhone X, iPhone 11, iPhone 13 Pro Max, iPhone 7, iPhone 13, iPhone 6s Plus, iPhone XS, iPhone SE (2nd generation), iPhone 8, iPhone 11 Pro, iPhone 12 Pro Max, iPhone 6, iPhone 13 Pro
4) 8.8.0 ( manually added)
5) Not used
6)
```
Crashed: dbrQueue
0  libsystem_kernel.dylib         0x7b78 __pthread_kill + 8
1  libsystem_pthread.dylib        0x73bc pthread_kill + 268
2  libsystem_c.dylib              0x2051c abort + 168
3  libsystem_malloc.dylib         0x1ca04 _malloc_put + 550
4  libsystem_malloc.dylib         0x1cc9c malloc_zone_error + 100
5  libsystem_malloc.dylib         0x170dc nanov2_allocate_from_block + 568
6  libsystem_malloc.dylib         0x16164 nanov2_allocate + 128
7  libsystem_malloc.dylib         0x16080 nanov2_malloc + 64
8  libsystem_malloc.dylib         0x6020 _malloc_zone_malloc + 156
9  libc++abi.dylib                0x138a0 operator new(unsigned long) + 32
10 DynamsoftBarcodeReader         0x3bde5c (Missing UUID x)
11 DynamsoftBarcodeReader         0x3c82b0 (Missing UUID x)
12 DynamsoftBarcodeReader         0x1a171c (Missing UUID x)
13 DynamsoftBarcodeReader         0x458ce0 (Missing UUID x)
14 DynamsoftBarcodeReader         0x45c574 (Missing UUID x)
15 DynamsoftBarcodeReader         0x65afbc (Missing UUID x)
16 DynamsoftBarcodeReader         0x65a494 (Missing UUID x)
17 DynamsoftBarcodeReader         0x679ac8 (Missing UUID x)
18 DynamsoftBarcodeReader         0x640708 (Missing UUID x)
19 DynamsoftBarcodeReader         0x635dbc (Missing UUID x)
20 DynamsoftBarcodeReader         0x6338d4 (Missing UUID x)
21 DynamsoftBarcodeReader         0x631d80 (Missing UUID x)
22 DynamsoftBarcodeReader         0x6303d0 (Missing UUID x)
23 DynamsoftBarcodeReader         0x3b1348 (Missing UUID x)
24 DynamsoftBarcodeReader         0x3a8f04 (Missing UUID x)
25 DynamsoftBarcodeReader         0x3a36a4 (Missing UUID x)
26 DynamsoftBarcodeReader         0x38525c (Missing UUID x)
27 DynamsoftBarcodeReader         0x387588 (Missing UUID x)
28 DynamsoftBarcodeReader         0x432fa4 (Missing UUID x)
29 DynamsoftBarcodeReader         0x3d7ee0 (Missing UUID x)
30 DynamsoftBarcodeReader         0x158b8 (Missing UUID x)
31 DynamsoftBarcodeReader         0x15184 (Missing UUID x)
32 Metro Turkiye                  0x34ca98 BarcodeReaderManager.decodeBarcode(_:) + 311 (BarcodeReaderManager.swift:311)
33 Metro Turkiye                  0x34d1c8 @objc BarcodeReaderManager.captureOutput(_:didOutput:from:) + 4314829256 (<compiler-generated>:4314829256)
34 AVFCapture                     0x1908c -[AVCaptureVideoDataOutput _processSampleBuffer:] + 336
35 AVFCapture                     0x3590 __47-[AVCaptureVideoDataOutput _updateRemoteQueue:]_block_invoke + 104
36 CMCapture                      0x14cfe4 __FigRemoteOperationReceiverCreateMessageReceiver_block_invoke + 304
37 CMCapture                      0x30e210 __rqReceiverSetSource_block_invoke + 256
38 libdispatch.dylib              0x3a2c _dispatch_client_callout + 20
39 libdispatch.dylib              0x6eec _dispatch_continuation_pop + 500
40 libdispatch.dylib              0x1a13c _dispatch_source_invoke + 1596
41 libdispatch.dylib              0x6dc8 _dispatch_continuation_pop + 208
42 libdispatch.dylib              0x6558 _dispatch_async_redirect_invoke + 588
43 libdispatch.dylib              0x15ec0 _dispatch_kevent_worker_thread + 1292
44 libsystem_pthread.dylib        0x10f4 _pthread_wqthread + 344
45 libsystem_pthread.dylib        0xe5c start_wqthread + 8
```

**Comment by rcunal** [comment]
Same problem occurs on android side too.

DBR version: 8.8.0 & 9.0.0
DCE: not used
Devices: wide variety

Redmi Note 8
Galaxy A72 (SM-A725F)
Redmi Note 9 Pro
Galaxy Note10+ (SM-N975F)
Galaxy M51 (SM-M515F)
Galaxy S10+ (SM-G975F)
Galaxy A70 (SM-A705FN)
Redmi 8
Redmi K20 (Mi 9T)
MI6 (MI 6)
Redmi Note 10 (M2101K7AG)
Redmi Note 9 (M2003J15SC)
GM 20 PRO (G501)
GM 20 (G301)
P20 lite (ANE-LX1)
P30 Pro (VOG-L29)
HUAWEI Y9 Prime 2019 (STK-L21)
Mate 20 lite (SNE-LX1)
Galaxy S22 (SM-S901E)
Galaxy Note10 (SM-N970F)
Galaxy Note8 (SM-N950F)
Galaxy J7 Neo (SM-J701F)
Galaxy S21 Ultra 5G (SM-G998B)
Galaxy S10e (SM-G970F)
Galaxy S8+ (SM-G955F)
Galaxy C5 Pro (SM-C5010)
Galaxy A52s 5G (SM-A528B)
Galaxy A52 (SM-A525F)
Galaxy A50 (SM-A505F)
Galaxy A21s (SM-A217F)
(SM-A125F)
Redmi Note 6 Pro (Redmi Note 6 Pro)
Redmi 7
Mate 10 lite (RNE-L21)
realme C21 (RMX3201)
realme C25s (RMX3195)
HUAWEI P smart 2019 (POT-LX1)
P13 Blue Max Pro
Mi Note 10 Lite
Mi A3
Mi A1
Mi 10
MI MAX 2
MI 8 Lite
Redmi Note 10 (M2103K19C)
Mi 11 Lite (M2101K9AG)
Redmi Note 10 Pro (M2101K6G
POCO X3 NFC (M2007J20CG)
Redmi 9C (M2006C3MG)
Redmi 9 (M2004J19C)
K61 (LM-Q630)
GM 21 Pro (G700)
GM 22 Plus (G512)
Reno2 Z (CPH1951)
A5 2020 (CPH1931)
Honor Play (COR-L29)
OnePlus Nord N10 5G (BE2029)
Xiaomi 11 Lite 5G NE (2109119DG)

Crash #1:

Crashed: Thread :  SIGABRT  0x0000000000000000
#00 pc 0x5f2b6 libc.so 
#01 pc 0x7c0703 libDynamsoftBarcodeReaderAndroid.so 
#02 pc 0x9643ce libDynamsoftBarcodeReaderAndroid.so 
#03 pc 0x1bed1 libc.so 
#04 pc 0xa7215 libc.so 
#05 pc 0xa6a83 libc.so 
#06 pc 0x1bed1 libc.so 
#07 pc 0x9643ce libDynamsoftBarcodeReaderAndroid.so 
#08 pc 0xa6971 libc.so 
#09 pc 0x7b51e7 libDynamsoftBarcodeReaderAndroid.so 
#10 pc 0x330fcf libDynamsoftBarcodeReaderAndroid.so 
#11 pc 0x9643ce libDynamsoftBarcodeReaderAndroid.so 
#12 pc 0x331fbf libDynamsoftBarcodeReaderAndroid.so 
#13 pc 0xaf55d libart.so 
#14 pc 0xad580 libart.so 
#15 pc 0x2f3cb3 libDynamsoftBarcodeReaderAndroid.so 
#16 pc 0xdc519 libart.so 
#17 pc 0x50ffd boot-core-libart.oat 

---------------------------------------------------------------------------
Crash #2:

main (native): tid=1 systid=23733 
#00 pc 0x4d81c libc.so 
#01 pc 0x52158 libc.so 
#02 pc 0xb5c68 libc.so 
#03 pc 0x8cdf50 libDynamsoftBarcodeReaderAndroid.so 
#04 pc 0x2d7e60 libDynamsoftBarcodeReaderAndroid.so 
#05 pc 0x30da70 libDynamsoftBarcodeReaderAndroid.so 
#06 pc 0x2bc08c libDynamsoftBarcodeReaderAndroid.so 
#07 pc 0x222244 libart.so 
#08 pc 0x218964 libart.so 
#09 pc 0x2851f0 libart.so 
#10 pc 0x3e70f0 libart.so 
#11 pc 0x3e37f8 libart.so 
#12 pc 0x761f50 libart.so 
#13 pc 0x203c14 libart.so 
#14 pc 0x3f47fa base.apk 
#15 pc 0x3d98e0 libart.so 
#16 pc 0x7464b4 libart.so 
#17 pc 0x222378 libart.so 
#18 pc 0x212520 libart.so 
#19 pc 0x12943e base.apk 
#20 pc 0x2124c4 libart.so 
#21 pc 0x1294dc base.apk 
#22 pc 0x2124c4 libart.so 
#23 pc 0x141848 base.apk 
#24 pc 0x2124c4 libart.so 
#25 pc 0xa69a2 base.apk 
#26 pc 0x2124c4 libart.so 
#27 pc 0x30da08 base.apk 
#28 pc 0x2124c4 libart.so 
#29 pc 0x3066e0 base.apk 
#30 pc 0x2124c4 libart.so 
#31 pc 0x306e04 base.apk 
#32 pc 0x2124c4 libart.so 
#33 pc 0x304438 base.apk 
#34 pc 0x2124c4 libart.so 
#35 pc 0x304c8a base.apk 
#36 pc 0x2124c4 libart.so 
#37 pc 0x3022f2 base.apk 
#38 pc 0x2124c4 libart.so 
#39 pc 0x2fff6e base.apk 
#40 pc 0x207c38c memfd:jit-cache 
#41 pc 0x206b348 memfd:jit-cache 
#42 pc 0x20347b4 memfd:jit-cache 
#43 pc 0x218be8 libart.so 
#44 pc 0x28520c libart.so 
#45 pc 0x3e70f0 libart.so 
#46 pc 0x3e1cec libart.so 
#47 pc 0x762d0c libart.so 
#48 pc 0x203994 libart.so 
#49 pc 0x4c8f14 framework.jar 
#50 pc 0x763268 libart.so 
#51 pc 0x203994 libart.so 
#52 pc 0x1bba8e framework.jar 
#53 pc 0x3d98e0 libart.so 
#54 pc 0x7464b4 libart.so 
#55 pc 0x222378 libart.so 
#56 pc 0x218be8 libart.so 
#57 pc 0x28520c libart.so 
#58 pc 0x627760 libart.so 
#59 pc 0x597a38 libart.so 
       at com.dynamsoft.dbr.BarcodeReader.nativeSetArg(Native method)
       at com.dynamsoft.dbr.BarcodeReader.initLicenseFromDLS(BarcodeReader.java:510)
       at com.metrotr.metroandroid.metrofast.MetroFastBarcodeScannerFragment.initDynamsoftScanner(MetroFastBarcodeScannerFragment.kt:229)
       at com.metrotr.metroandroid.metrofast.MetroFastBarcodeScannerFragment.onCreateFinished(MetroFastBarcodeScannerFragment.kt:63)
       at com.metrotr.metroandroid.metrofast.MetroFastShoppingListFragment.onCreateFinished(MetroFastShoppingListFragment.kt:70)
       at com.metrotr.metroandroid.BaseFragment.onViewCreated(BaseFragment.kt:55)
       at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:3019)
       at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:551)
       at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
       at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1840)
       at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1758)
       at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1701)
       at androidx.fragment.app.FragmentManager.run(FragmentManager.java:488)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8633)
       at java.lang.reflect.Method.invoke(Native method)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1133)

Crash #3:

Fatal Exception: java.lang.Error: com.dynamsoft.dbr.BarcodeReaderException: Unknown error.
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1173)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:764)

Caused by com.dynamsoft.dbr.BarcodeReaderException: Unknown error.
       at com.dynamsoft.dbr.BarcodeReader.nativeDecodeBuffer(BarcodeReader.java)
       at com.dynamsoft.dbr.BarcodeReader.decodeRawImage(BarcodeReader.java:2250)
       at com.dynamsoft.dbr.BarcodeReader.decodeBuffer(BarcodeReader.java:809)
       at com.metrotr.metroandroid.metrofast.MetroFastBarcodeScannerFragment$BarcodeAnalyzer.analyze(MetroFastBarcodeScannerFragment.kt:192)
       at androidx.camera.core.ImageAnalysis.lambda$setAnalyzer$ImageAnalysis(ImageAnalysis.java:367)
       at androidx.camera.core.-$$Lambda$ImageAnalysis$SK4XUin3LEuOy83vccLKZvMIwq8.analyze(:4)
       at androidx.camera.core.ImageAnalysisAbstractAnalyzer.lambda$analyzeImage$ImageAnalysisAbstractAnalyzer(ImageAnalysisAbstractAnalyzer.java:121)
       at androidx.camera.core.-$$Lambda$ImageAnalysisAbstractAnalyzer$V6Gtux2-h8spda8vf4wFnIZntYI.run(:8)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:764)

**Comment by burakcolakappcent** [comment]
@Dynamsoft-Jerson

**Comment by burakcolakappcent** [comment]
@Dynamsoft-Jerson why did you close this task, do you any working improvement or prove that this crashed resolved ?

**Comment by Dynamsoft-Jerson** [comment]
@burakcolakappcent Yes, we have fixed a c++ crash bug on DBR v9.0.2. Have you tried it?
